pkg/api/v20240812preview/openshiftcluster_validatestatic.go

```diff
@@ -412,3 +420,49 @@ func (sv openShiftClusterStaticValidator) validateDelta(oc, current *OpenShiftCl
 
 	return nil
 }
+
+func (sv openShiftClusterStaticValidator) validatePlatformWorkloadIdentityProfile(path string, pwip *PlatformWorkloadIdentityProfile) error {
+	// PlatformWorkloadIdentityProfile being empty is acceptable
+	if pwip == nil {
+		return nil
+	}
+
+	// Validate the PlatformWorkloadIdentities
+	for n, p := range pwip.PlatformWorkloadIdentities {
```

Comment:
@cadenmarchese @rajdeepc2792, should we be validating `ClientID` and `ObjectID` within this loop too? I think we determined a while back that the RP will be able to fill these two fields in on its own given the MI's resource ID, but I'm still a bit unsure - even if we do fill in these fields on behalf of the cx, is that magic going to happen in the RP or the CLI? If it happens in the CLI, we should add the two fields to static validation.

@tsatam , maybe you have some thoughts on this too?

Reply:
CLI is planning to accept only resourceIDs - while we _could_ infer client/objectIDs for a given MI resource ID CLI-side, it would be better if it was handled RP-side for consistency across non-CLI usage as well as correctness guarantees.

Reply:
Okay, then we can leave them out of static validation, since it wouldn't make sense to validate a field that the CLI won't be passing to the RP.

No changes needed here @yithian , and I'll resolve the conversation.